File: AppSettings.cs
using System.Text.Json.Serialization;

namespace LANchat
{
    [System.Serializable]
    public class AppSettings : Connection.ILoadable
    {
        public string Username;
        public ushort Port;

        [JsonIgnore]
        public bool IsDefaultPort
        {
            get
            {
                return Port == GlobalDefs.DefaultPort;
            }
        }

        public AppSettings() { }

        [JsonConstructor]
        private AppSettings(string username, ushort port)
        {
            Username = username;
            Port = port;
        }

        public static AppSettings DefaultSettings() => new AppSettings()
        {
            Username = System.Environment.UserName,
            Port = GlobalDefs.DefaultPort
        };

        public override string ToString() => $"User '{Username}' on port {Port}";
    }
}

File: app/appsettings.cpp
#include "appsettings.h"
#include "src/globaldefs.h"
#include "exceptions.h"
using namespace Exceptions;

#include <QDir>

AppSettings::AppSettings() { }

AppSettings AppSettings::DefaultSettings()
{
    AppSettings a;
    a.Username = QDir::home().dirName();
    a.Port = GlobalDefs::DefaultPort;
    return a;
}

bool AppSettings::IsDefaultPort() const
{
    return Port == GlobalDefs::DefaultPort;
}

QMap<QString, QString>& AppSettings::Serialize()
{
    QMap<QString, QString>* varValueMap = new QMap<QString, QString> { { "ClassName", "AppSettings" } };
    varValueMap->insert("Username", Username);
    varValueMap->insert("Port", QString::number(Port));
    return *varValueMap;
}

void AppSettings::Deserialize(QMap<QString, QString>& varValueMap)
{
    if (varValueMap["ClassName"] != "AppSettings")
    {
        throw CommonException("Field 'ClassName' is not equals to 'AppSettings'");
    }
    Username = varValueMap["Username"];
    Port = varValueMap["Port"].toUInt();
}

QString AppSettings::ToString()
{
    return QString::asprintf("'%s' on port %u", Username.toStdString().c_str(), Port);
}
